Fix an issue with attendee saving and add some more flexibility to service handler

dcaliste opened the pull request:
While working on attendee support in nemo-qml-plugin-calendar, I noticed that it's not convenient to default to the invitation plugin in tests. I'm thus adding a commit here so mKCal can look for plugins in a directory that can be given with an environment variable. Then tests can use this variable to load ad hoc plugins from the test directory without poluting the system directories.

I also noticed that saving the organizer as an attendee was not optimal when the organizer is actually in the attendee list, because his role, status and RSVP were resetted to default values. I've added a test demonstrating the problem and I added a fix in the saving code : if the organizer is in the attendee list, use it's attendee definition instead of creating a new one.

@pvuorela , not in a hurry, when you have a moment to give a look. Thank you.

pvuorela commented:
> While working on attendee support in nemo-qml-plugin-calendar, I noticed that it's not convenient to default to the invitation plugin in tests. I'm thus adding a commit here so mKCal can look for plugins in a directory that can be given with an environment variable. Then tests can use this variable to load ad hoc plugins from the test directory without poluting the system directories.

Makes sense. Could you add some mention of the unit testing to the commit message too.

On that service(serviceId), was there some specific need in mind as it's an api addition?

pvuorela commented:
> [mkcal] Don't trash origanizer role and status.

Small typo there ^

One thing though that if I some day manage to get back to the sql scheme adjustment I was planning, the restored attendee & organizer should be more or less the same that got inserted.

dcaliste commented:
> Could you add some mention of the unit testing to the commit message too.

Done, I've added an explicit mention of the test usage in the commit message.

> On that service(serviceId), was there some specific need in mind as it's an api addition?

In the test plugin in nemo-qml-plugin-calendar, I need to access data inside the plugin (that are stored depending on which function has been called with what arguments). Like that, I can test if the functions were called properly and with the expected arguments.

> Small typo there ^

Thanks, fixed.

> One thing though that if I some day manage to get back to the sql scheme adjustment I was planning, the restored attendee & organizer should be more or less the same that got inserted.

I think the changes here won't disable this. The given organizer and attendees are almost restored as they were provided. There are two differences at the moment :
- one due to SQlite scheme limitation : on reread, organizer is always in the attendee list, whatever his status before,
- one new here, introduced in this PR : on reread, organizer name is overwritten by attendee name with the same email, when organizer was in the attendee list. It is left untouched otherwise.

About the second point, the RFC5545 is not saying anything on the organizer name and attendee name sharing the same email. They could be different. But due to our scheme limitation, we need to make a choice here : organizer will be added in the attendee list, so what should be his name in case it was actually already in the attendee list ? Or seen reversely, why should we overwrite the attendee name that share the same email as the organizer with the organizer name as we are doing now ? I preferred the first solution because it is consistent with the attendee list, which is what is exposed to QML.

pvuorela commented:
Yea guess this is the best we can do before the scheme changes and when that happens these parts of the code need to be revisited anyway, e.g. organizer not written to sql anymore with insertAttendee(). The "Organizer is not in the attendee list but mKCal is adding him." part of the test would be also removed at that point.

So LGTM.
